commit dcdf8d51e736382c3f11542830b656593b73e299
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:36 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-conditional-ccachedir-creation.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 50555ed99..9c3b1cead 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -429,15 +429,17 @@ proc portconfigure::configure_start {args} {
     if {${configure.ccache}} {
         global ccache_dir ccache_size macportsuser
         # Create ccache directory with correct permissions with root privileges
-        elevateToRoot "configure ccache"
-        if {[catch {
+        if {![file exists ${ccache_dir}] || [file type ${ccache_dir}] ne "directory"} {
+            try {
+                elevateToRoot "Set up ccache directory `${ccache_dir}`"
                 file mkdir ${ccache_dir}
                 file attributes ${ccache_dir} -owner ${macportsuser} -permissions 0755
-            } result]} {
-            ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $result"
-            set configure.ccache no
+            } catch {{*} eCode eMessage} {
+                ui_warn "ccache_dir ${ccache_dir} could not be created; disabling ccache: $eMessage"
+                set configure.ccache no
+            }
+            dropPrivileges
         }
-        dropPrivileges
 
         # Initialize ccache directory with the given maximum size
         if {${configure.ccache}} {
